dist/js/multiselect.js

```diff
@@ -1,755 +1,1812 @@
+                    $optionOrOptgroup.data(DATA_POSITION, newPosition);
+                }
+                return $optionOrOptgroup;
+            }
+
+            /**
+             * Either prepends the given search filter html to the target select element
+             * just uses an existing input for the filter. Also stores which
+             * selects should be affected by the filter.
+             * @param {jQuery} $targetSelect - the select element where the filter should be prepended
+             * @param {string|jQuery} searchFilterHtmlOrElement - the html for the search filter or an existing jQuery element
+             * @param {jQuery} $filteredSelects - the select elements that should be filtered with this input
+             * @returns {FilterRelation} - the filter/selects relation
+             */
+            function prependSearchFilter($targetSelect, searchFilterHtmlOrElement, $filteredSelects) {
+                /** The created search filter element
+                 * @type {jQuery} */
+                var $searchFilter = $();
+                if (typeof searchFilterHtmlOrElement === "string") {
+                    $searchFilter = $(searchFilterHtmlOrElement);
+                    $targetSelect.before($searchFilter);
+                } else if (searchFilterHtmlOrElement instanceof $) {
+                    //noinspection JSValidateTypes
+                    $searchFilter = searchFilterHtmlOrElement;
+                }
+                return {
+                    $filterInput: $searchFilter,
+                    $filteredSelects: $filteredSelects
+                };
+            }
+
+            /**
+             * The old filter function stored for reference.
+             * @param {string} filterValue - the value options must match
+             * @param {jQuery} $select - the select to filter
+             */
+            function oldFilterOptions(filterValue, $select) {
+                // options to show
+                extendedShow($select.find('option:search("' + filterValue + '")'));
+                // options to hide
+                extendedHide($select.find('option:not(:search("' + filterValue + '"))'));
+                // optgroups to hide
+                extendedHide($select.find('option.hidden').parent('optgroup').not($(":visible").parent()));
+                // optgroups to show
+                extendedShow($select.find('option:not(.hidden)').parent('optgroup'));
+            }
+
+            /**
+             * Filters the given selects using the filter value
+             * to check which options should be filtered.
+             * @param {string} filterValue - the value to use for choosing options to filter
+             * @param {jQuery} $selectsToFilter - the select elements that should be filtered using the value
+             */
+            function applyFilter(filterValue, $selectsToFilter) {
+                if (!filterValue || filterValue == "") {
```

Comment:
This is my overly wordy filter function. Probably way too many jQuery collections, I don't know.